### PR TITLE
Add dedicated product details page

### PR DESCRIPTION
**Pull request by bogminic**
### Description

Hi team! 👋 As promised, I've created the PR with the product details page after a lot of refinement. Please take a look!

I replaced the product-details modal with a standalone, lazily-loaded `product/:id` page and migrated the order-history review navigation to it.

### Key changes

**ProductPageComponent** — New standalone product page with:

**Sections:**
1. Product Info — Responsive two-column card layout with quantity stepper and add-to-basket button
2. Reviews — Expandable panel with add/edit/like functionality and deep-link support (`?goto-reviews=true`)
3. Related Products — Responsive row using `ResizeObserver` + computed `gridColumns`

**Visual Enhancements:**
- Strikethrough on deluxe pricing
- Centered bonus-points badge on the crown

**User Experience:**
- Loading spinners for product, reviews, and related products
- "Product not found" message for missing products

**Breaking Changes:**
- Removed `ProductDetailsComponent`
- Order history now navigates to the product page

**Testing:**
- Unit tests for spinners, basket behavior, bonus points, pricing, and navigation

### AI Tool Disclosure

- [ ] My contribution does not include any AI-generated content
- [x] My contribution includes AI-generated content, as disclosed below:
    - AI Tools: `Visual Studio Code AI Chat`
    - LLMs and versions: `DeepSeek V4 Pro`
    - Prompts: `I asked to replace the product-details modal with a dedicated lazy product/:id page and then iteratively refined it — readable markup, Material card, loading spinners (product, reviews, related), a centered and resized bonus-points badge, a "Product not found" message, spinner/edge-case tests, a minimal gridColumns+computed related-products layout with a ResizeObserver`

### Affirmation

- [x] My code follows the [CONTRIBUTING.md](https://github.com/juice-shop/juice-shop/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

**Comment by bkimminich** [comment]
The buttons and text are disturbingly oversized compared to all other screens in Juice Shop... :-D

image

**Comment by bkimminich** [comment]
Same on medium size screens:

image

And mobile view is really bad with having to 100% scroll down past the image:

image

File: frontend/src/app/product-page/product-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+  private scrollToReviewsSection (element: HTMLElement): void {
+    requestAnimationFrame(() => {
+      requestAnimationFrame(() => {
+        element.scrollIntoView({ behavior: 'smooth', block: 'start' })
+      })
+    })
+  }
+
+  private observeGrid (grid: HTMLElement): void {
```
Comment (J12934) [inline frontend/src/app/product-page/product-page.component.ts]: I think to have the code be simpler I'd rather have a fixed number or max number of related products (e.g. 8) and have the row have a horizontal scroll.

Then we don't need resize observer logic.

Also right now with the current logic the related product on mobile always just shows one product, which doesn't seem right to me

**Comment by bogminic** [comment]
> The buttons and text are disturbingly oversized compared to all other screens in Juice Shop... :-D
> 
> image

I've made the button smaller, but I'm maintaining it at a larger size than the related products button. Does this proportion work?

product-page-desktop

**Comment by bogminic** [comment]
> Same on medium size screens:
> 
> image
> And mobile view is really bad with having to 100% scroll down past the image:
> 
> image

Shrinking the button works OK-is on medium screens. I'm also using the same approach to lock the product images at a 16:9 aspect ratio on smaller devices. 
I have the same question—what do you think?

**Medium**
Screenshot From 2026-09-01 10-09-21

**Mobile**
Screenshot From 2026-09-01 10-12-29
